packages/console/src/components/RoleScopesTransfer/components/SourceScopesBox/index.tsx

```diff
@@ -0,0 +1,120 @@
+import type { ResourceResponse, Scope, ScopeResponse } from '@logto/schemas';
+import { conditional } from '@silverhand/essentials';
+import type { ChangeEvent } from 'react';
+import { useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import useSWR from 'swr';
+
+import Search from '@/assets/images/search.svg';
+import type { DetailedResourceResponse } from '@/components/RoleScopesTransfer/types';
+import TextInput from '@/components/TextInput';
+import * as transferLayout from '@/scss/transfer.module.scss';
+
+import ResourceItem from '../ResourceItem';
+import * as styles from './index.module.scss';
+
+type Props = {
+  roleId?: string;
+  selectedScopes: ScopeResponse[];
+  onChange: (value: ScopeResponse[]) => void;
+};
+
+const SourceScopesBox = ({ roleId, selectedScopes, onChange }: Props) => {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const { data = [] } = useSWR<ResourceResponse[]>('/api/resources?includeScopes=true');
+  const { data: roleScopes = [] } = useSWR<Scope[]>(roleId && `/api/roles/${roleId}/scopes`);
+
+  const [keyword, setKeyword] = useState('');
+
+  const handleSearchInput = (event: ChangeEvent<HTMLInputElement>) => {
+    setKeyword(event.target.value);
```

Comment:
debounce?

Reply:
This search happens locally and it don't need debounce.